PhysicsTools/PatAlgos/plugins/JPTJetSlimmer.cc

```diff
@@ -0,0 +1,97 @@
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "DataFormats/Common/interface/RefVector.h"
+
+#include "CommonTools/UtilAlgos/interface/StringCutObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/SingleObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/ObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/SingleElementCollectionSelector.h"
+
+#include "DataFormats/JetReco/interface/JPTJet.h"
+#include "DataFormats/JetReco/interface/JPTJetCollection.h"
+#include "DataFormats/JetReco/interface/CaloJet.h"
+#include "DataFormats/JetReco/interface/CaloJetCollection.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/Common/interface/RefToPtr.h"
+#include <vector>
+
+class JPTJetSlimmer : public edm::stream::EDProducer<> {
+public:
+  JPTJetSlimmer(edm::ParameterSet const& params)
+      : srcToken_(consumes<edm::View<reco::JPTJet> >(params.getParameter<edm::InputTag>("src"))),
+        srcCaloToken_(consumes<edm::View<reco::CaloJet> >(params.getParameter<edm::InputTag>("srcCalo"))),
+        cut_(params.getParameter<std::string>("cut")),
+        selector_(cut_) {
+    produces<reco::JPTJetCollection>();
+    produces<reco::CaloJetCollection>();
+  }
+
+  ~JPTJetSlimmer() override {}
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  void produce(edm::Event& iEvent, const edm::EventSetup& iSetup) override {
+    auto jptJets = std::make_unique<reco::JPTJetCollection>();
+    auto caloJets = std::make_unique<reco::CaloJetCollection>();
+
+    edm::RefProd<reco::CaloJetCollection> pOut1RefProd = iEvent.getRefBeforePut<reco::CaloJetCollection>();
+    edm::Ref<reco::CaloJetCollection>::key_type idxCaloJet = 0;
+
+    auto const& h_calojets = iEvent.get(srcCaloToken_);
+    auto const& h_jets = iEvent.get(srcToken_);
+
+    for (auto const& ijet : h_jets) {
+      if (selector_(ijet)) {
+        // Add specific : only reference to CaloJet collection. It is necessary for
+        // recalibration JPTJet at MiniAod.
+        const edm::Ptr<reco::CaloJet>& rawcalojet = ijet.getCaloJetRef();
+        int icalo = -1;
+        int i = 0;
+        for (auto const& icjet : h_calojets) {
+          double dr2 = deltaR2(icjet, *rawcalojet);
+          if (dr2 <= 0.001) {
```

Comment:
what's the motivation behind the hardcoded value of 0.001? Are you checking the equivalence of two jets? Can this fail in rare cases?

Reply:
No, this is  either the same jets (if JPT calo seed is presented in slimmedCalojet collection) or it is some completely other jet.  From the other side jets are separated on the level of dr2 >= ~0.1. Thus, it's safe. It can be dr2 == 0, but as we work with doubles I would prefer to put less then some small value.

Reply:
but what motivates the exact value? in general we try to avoid arbitrary hardcoded "sameness" values.

Reply:
Coming back on this point, @kodolova, is there a way to use directly `rawcalojet` (`edm::RefToBase`) to search for the ref of `reco::CaloJet` instead of this `deltaR`?

Reply:
I do not think so, because JPT has RefToBase to reco::Calojet collection. We look for the reco::Calojet that is kept at MiniAOD: slimmedCalojets<---It is another Collection. I suspect that RefToBase from one Collection can not be used for the completely different Collection.